ref-exchange/src/account_deposit.rs

```diff
@@ -5,29 +5,100 @@ use std::convert::TryInto;
 
 use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
 use near_sdk::json_types::{ValidAccountId, U128};
-use near_sdk::{assert_one_yocto, env, near_bindgen, AccountId, Balance, PromiseResult};
+use near_sdk::{
+    assert_one_yocto, env, near_bindgen, AccountId, Balance, PromiseResult, StorageUsage,
+};
 
 use crate::utils::{ext_fungible_token, ext_self, GAS_FOR_FT_TRANSFER};
 use crate::*;
 
-const MAX_ACCOUNT_LENGTH: u128 = 64;
-const MIN_ACCOUNT_DEPOSIT_LENGTH: u128 = MAX_ACCOUNT_LENGTH + 16 + 4;
+const U128_STORAGE: StorageUsage = 16;
+/// bytes length of u64 values
+const U64_STORAGE: StorageUsage = 8;
+/// bytes length of u32 values. Used in length operations
+const U32_STORAGE: StorageUsage = 4;
+/// max length of account id * 8 (1 byte)
+const ACC_ID_STORAGE: StorageUsage = 64 * 8;
+// 64 = max account name length
+
+// ACC_ID: the Contract accounts map
+// + U128_STORAGE: near_amount storage
+// + U32_STORAGE: tokens HashMap length
+// + U64_STORAGE: storage_used
+// + 2: version
+pub const INIT_ACCOUNT_STORAGE: StorageUsage =
+    ACC_ID_STORAGE + U128_STORAGE + U32_STORAGE + 2 * U64_STORAGE + 2;
+
+// NEAR native token. This is not a valid token ID. HACK: NEAR is a native token, we use the
+// empty string we use it to reference not existing near account.
+// pub const NEAR: AccountId = "".to_string();
+
+#[derive(BorshDeserialize, BorshSerialize)]
+pub enum AccountDeposit {
```

Comment:
I like more VersionedAccount and Account for the current version.
This way everywhere in the code except in the places where it's used - it's just Account. And VersionedAccount is where need to store/load/upgrade

Reply:
So, you want to have a `LegacyAccount` and `Account`? What if we will do another migration? I think having a consecutive versions in struct name is more clear. We will mark all old version _Legacy_ in the comments.